tests/lib/prepare-restore.sh

```diff
@@ -634,6 +634,11 @@ prepare_suite_each() {
         "$TESTSTOOLS"/cleanup-state pre-invariant
     fi
     tests.invariant check
+
+    SNAP_MOUNT_DIR="$(os.paths snap-mount-dir)"
+    if os.query is-focal || os.query is.fedora; then
+        inotifywait /root /var/lib/snapd /var/snap /home "$SNAP_MOUNT_DIR" -d -m -r -e CREATE -e DELETE --exclude "($PWD/task.yaml|$PWD/.*/.*.snap|$PROJECT_PATH/tests/lib/snaps/.*/.*.snap)" -o /tmp/fs.output
```

Comment:
we may want to monitor /etc too, but that can be tricky when the test installs new packages

Reply:
I'll add it, in case it causes trouble, I'll remove it.

Reply:
I just run with /etc and the following paths are breaking tests:
File not deleted /etc/systemd/system/var-lib-snapd-snap-core-11378.mount
File not deleted /etc/systemd/system/multi-user.target.wants/var-lib-snapd-snap-core-11378.mount
File not deleted /etc/environment
File not deleted /etc/systemd/system/multi-user.target.wants/var-lib-snapd-snap-core-11378.mount
Dir not deleted /etc/systemd/system/snapd.service.d

Reply:
Looks like core 11378 could have been the revision the system came with and it got restored after the test finished?

> Dir not deleted /etc/systemd/system/snapd.service.d
isn't this something the test setup does?